From 41a09d2b52a8960606f61ba2f8b51e8f029d84bc Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Tue, 24 Jan 2023 12:55:30 +0100
Subject: [PATCH 8/8] ENH: Add GNUInstallDirs to installation paths

This modifies the Slicer installation paths to include the CMake
GNUInstallDirs installation
directories (https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html).
---
 CMake/SlicerDirectories.cmake | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMake/SlicerDirectories.cmake b/CMake/SlicerDirectories.cmake
index 10785b6fd1..b27a2aa350 100644
--- a/CMake/SlicerDirectories.cmake
+++ b/CMake/SlicerDirectories.cmake
@@ -55,11 +55,11 @@ endif()
 # Slicer relative directories
 #-----------------------------------------------------------------------------
 # for build tree
-set(Slicer_BIN_DIR "bin")
-set(Slicer_LIB_DIR "lib/${Slicer_MAIN_PROJECT_APPLICATION_NAME}-${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
-set(Slicer_INCLUDE_DIR "include/${Slicer_MAIN_PROJECT_APPLICATION_NAME}-${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
-set(Slicer_SHARE_DIR "share/${Slicer_MAIN_PROJECT_APPLICATION_NAME}-${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
-set(Slicer_LIBEXEC_DIR "libexec/${Slicer_MAIN_PROJECT_APPLICATION_NAME}-${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
+set(Slicer_BIN_DIR "${CMAKE_INSTALL_BINDIR}")
+set(Slicer_LIB_DIR "${CMAKE_INSTALL_LIBDIR}")
+set(Slicer_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDE_DIR}/${Slicer_MAIN_PROJECT_APPLICATION_NAME}/${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
+set(Slicer_SHARE_DIR "${CMAKE_INSTALL_SHARE_DIR}/${Slicer_MAIN_PROJECT_APPLICATION_NAME}/${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
+set(Slicer_LIBEXEC_DIR "${CMAKE_INSTALL_LIBEXEC_DIR}/${Slicer_MAIN_PROJECT_APPLICATION_NAME}-${Slicer_VERSION_MAJOR}.${Slicer_VERSION_MINOR}")
 set(Slicer_ITKFACTORIES_DIR "${Slicer_LIB_DIR}/ITKFactories")
 set(Slicer_QM_DIR "${Slicer_BIN_DIR}/translations")
 
-- 
2.34.1

